lib/tests/test_mut_repo.rs

```diff
@@ -676,3 +676,76 @@ fn test_remove_wc_commit_previous_discardable() {
     mut_repo.rebase_descendants(&settings).unwrap();
     assert!(!mut_repo.view().heads().contains(old_wc_commit.id()));
 }
+
+#[test]
+fn test_reparent_descendants() {
+    // Test that MutableRepo::reparent_descendants() reparents descendants of
+    // rewritten commits without altering their content.
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    let mut tx = repo.start_transaction(&settings);
+    let mut graph_builder = CommitGraphBuilder::new(&settings, tx.repo_mut());
+    let commit_a = graph_builder.initial_commit();
+    let commit_b = graph_builder.initial_commit();
+    let commit_child_a_b = graph_builder.commit_with_parents(&[&commit_a, &commit_b]);
+    let commit_grandchild_a_b = graph_builder.commit_with_parents(&[&commit_child_a_b]);
+    let commit_child_a = graph_builder.commit_with_parents(&[&commit_a]);
+    let commit_child_b = graph_builder.commit_with_parents(&[&commit_b]);
+    let mut_repo = tx.repo_mut();
+    for (bookmark, commit) in [
+        ("b", &commit_b),
+        ("child_a_b", &commit_child_a_b),
+        ("grandchild_a_b", &commit_grandchild_a_b),
+        ("child_a", &commit_child_a),
+        ("child_b", &commit_child_b),
+    ] {
+        mut_repo.set_local_bookmark_target(bookmark, RefTarget::normal(commit.id().clone()));
+    }
+    let repo = tx.commit("test");
+
+    // Rewrite "commit_a" by removing its content.
+    let mut tx = repo.start_transaction(&settings);
+    let mut_repo = tx.repo_mut();
+    let empty_commit_tree_id = mut_repo.store().empty_merged_tree_id().clone();
+    mut_repo
+        .rewrite_commit(&settings, &commit_a)
+        .set_tree_id(empty_commit_tree_id)
```

Comment:
Maybe better to use another random tree (or file content updates) here.

If a file deleted from both merge sides, the result is deletion. This test passes even if we did `rebase_descendants()`.

Reply:
Good catch. Fixed.